tests/functional/test_install_compat.py

```diff
@@ -32,7 +32,7 @@ def test_debian_egg_name_workaround(script):
     # so even if this test runs on a Debian/Ubuntu system with broken
     # setuptools, since our test runs inside a venv we'll still have the normal
     # .egg-info
-    assert egg_info in result.files_created, \
+    result.did_create(egg_info), \
         "Couldn't find {egg_info}".format(**locals())
```

Comment:
`result.did_create(x), message` wouldn't work, right -- Either message needs to be passed as a parameter, or a better option IMHO would be to let `result.did_create` return a boolean, and assert the correctness in the test function itself. This would also improve it's readability:

```python
assert result.did_create(x), message
```

@pradyunsg would love to hear what you think on this.

Reply:
I also support the way @RJ722 suggested. That does seem straightforward. :)

Reply:
The thing is, we'd want to *always* have the message include `str(result)`, since it contains useful debugging information. Part of the point of making this change, is to get rid of needing to specify the message *every time*.

I think we'd likely want to add a message keyword-only argument to result, so that we can add to the default error message:

    result.did_create(X, message=message)

That should do:

    assert X in self.created, "\n".join(message, str(self))